src/ekat/CMakeLists.txt

```diff
@@ -7,6 +7,15 @@ EkatConfigFile (${CMAKE_CURRENT_SOURCE_DIR}/ekat_config.h.in
                 ${CMAKE_CURRENT_BINARY_DIR}/ekat_config.h
                 F90_FILE ${CMAKE_CURRENT_BINARY_DIR}/ekat_config.f)
 
+
+# Generate the program that generates the physics constants files
+set(EKAT_SRC_ROOT ${PROJECT_SOURCE_DIR})
+set(CONSTS_INPUT_FILE ${PROJECT_SOURCE_DIR}/src/ekat/physics/generate_constants_main.cpp.in)
+set(CONSTS_OUTPUT_FILE ${CMAKE_CURRENT_BINARY_DIR}/generate_constants_main.cpp)
+configure_file(${CONSTS_INPUT_FILE} ${CONSTS_OUTPUT_FILE} @ONLY)
+
+# TO DO: Build and run that program to generate the constants source files (.hpp, .cpp, .f90)
```

Comment:
This can be achieved via cmake's `add_custom_target` and `add_custom_command` (see [here](https://github.com/E3SM-Project/scream/blob/a77d39ffc3d02b6970792adef5b72efedaa27044/components/scream/src/physics/rrtmgp/tests/CMakeLists.txt#L13) for an example of how to use them to get what we want).

If you are not familiar with them and don't have time to play with cmake, I can contribute.

Reply:
I foresee something like this

```
macro (EkatCreatePhysConstantsSource)
  set(options Fortran CXX)
  set(args1v)
  set(argsMv)
  cmake_parse_arguments(ECPCS "${options}" "${args1v}" "${argsMv}" ${ARGN})

 # build generator code
 add_executable (...)
 
 if (ECPCS_Fortran) 
   # run with -f90 flag
 endif()
  if (ECPCS_CXX) 
   # run with -hpp and -cpp flags
 endif()
endmacro ()
```